kolibri/plugins/coach_tools/api.py

```diff
@@ -22,7 +22,18 @@ class KolibriReportPermissions(permissions.BasePermission):
 
     # check if requesting user has permission for collection or user
     def has_permission(self, request, view):
-        return request.user.can_read(get_collection_or_user(view.kwargs))
+        collection_kind = view.kwargs.get('collection_kind', None)
+        collection_id = view.kwargs.get('collection_id', None)
+        user_pk = view.kwargs.get('pk', None)
+
+        if any(collection_kind in kind for kind in collection_kinds.choices):
+            perm_check_obj = Collection.objects.get(pk=collection_id)
+        elif collection_id:
+            perm_check_obj = FacilityUser.objects.get(pk=collection_id)
+        else:  # check necessary for usersummary endpoint
+            perm_check_obj = FacilityUser.objects.get(pk=user_pk)
+
+        return request.user.can_read(perm_check_obj)
```

Comment:
We actually want to do a role check here, not just check if the object is readable (because, for example, a member of a collection can read the collection object, but not necessarily read data for other members). The current user should have a coach or admin role for the target object.